s4e-backend/src/main/java/pl/cyfronet/s4e/controller/AppUserController.java

```diff
@@ -64,14 +64,26 @@
     @PostMapping(value = "/register", consumes = APPLICATION_JSON_VALUE)
     public void register(
             @RequestBody @Valid RegisterRequest registerRequest,
-            HttpServletRequest request
-    ) throws RecaptchaException {
+            HttpServletRequest request,
+            @RequestParam(required = false) String token
+    ) throws RecaptchaException, NotFoundException {
         val email = registerRequest.getEmail();
 
         validateRecaptcha(request);
 
         try {
             appUserService.register(registerRequest);
+
+            // IMPORTANT!!! not found invitation shouldn't return 404 for correct user registration
+            try {
+                if (token != null) {
+                    invitationService.confirmInvitation(email, token);
+                    val confirmInvitationEvent = new OnConfirmInvitationEvent(token, LocaleContextHolder.getLocale());
+                    eventPublisher.publishEvent(confirmInvitationEvent);
+                }
+            }
+            catch (NotFoundException e) {}
```

Comment:
Catch Exception, and not NFE, as the called methods may throw runtime exceptions too.
I'd also log the thrown exceptions at level info.

Also, remove the newline between the closing try bracket and `catch`.